apps/website/app/blog/page.tsx

```diff
@@ -0,0 +1,63 @@
+import fs from "fs";
+import path from "path";
+import matter from "gray-matter";
+import Link from "next/link";
+import { BlogSchema, type Blog } from "./schema";
+
+async function getAllBlogs(): Promise<Blog[]> {
+  const blogDirectory = path.join(process.cwd(), "app/blog/posts");
+  const files = fs.readdirSync(blogDirectory);
+
+  return files.map((filename) => {
+    const filePath = path.join(blogDirectory, filename);
+    const fileContent = fs.readFileSync(filePath, "utf-8");
+    const { data } = matter(fileContent);
+
+    const validatedData = BlogSchema.parse(data);
+
+    return {
+      slug: filename.replace(".md", ""),
+      ...validatedData,
+    };
+  });
+}
+
+export default async function BlogIndex() {
+  const blogs = await getAllBlogs();
+  return (
+    <div className="flex-1 bg-gray-50">
+      <div className="mx-auto max-w-6xl space-y-12 px-6 py-12">
+        <div className="rounded-xl bg-white p-8 shadow-md">
+          <div className="mb-8">
+            <h1 className="text-4xl font-bold text-primary">All Updates</h1>
+          </div>
+          <div>
+            <ul className="space-y-6">
+              {blogs.map((blog) => (
```

Comment:
for Production we should probably add something like

```typescript
           {blogs.length === 0 ? (
              <p className="py-8 text-center text-gray-500">
                No updates yet! Check back soon. 😊
              </p>
            ) : (
```

and delete the example posts